### PR TITLE
[OTel] unittest coverage improvement

### PR DESCRIPTION
**Pull request by CagriYonca**
I worked on the modules below, increased existing unittests and created non-existing:
- src/instana/collector/base.py (100% coverage)
- src/instana/collector/helpers/runtime.py (97%)
- src/instana/collector/host.py (96%)
- src/instana/collector/utils.py (100%)
- src/instana/propagators/base_propagator.py (96%)
- src/instana/sampling.py (100%)
- src/instana/util/traceutils.py (100%)

### CODE REVIEW

File: src/instana/propagators/base_propagator.py [hunk 1]
```diff
@@ -232,15 +251,17 @@ def __determine_span_context(
...
 
         if traceparent:
             ctx_traceparent = traceparent
             ctx_tracestate = tracestate
 
         return SpanContext(
-            trace_id=ctx_trace_id,
-            span_id=ctx_span_id,
+            trace_id=int(ctx_trace_id),
+            span_id=int(ctx_span_id),
```
Comment (GSVarsha) [inline src/instana/propagators/base_propagator.py:264]: Why is this change needed?

Comment (CagriYonca) [reply src/instana/propagators/base_propagator.py:264]: Because span context is waiting for 2 int values for trace id and span id, and since we use get_instana_ancestor to provide these values, they may be strings (as you can see in the test case below)

image

Comment (pvital) [reply src/instana/propagators/base_propagator.py:264]: Don't change that at this point. 
If you read the code, `ctx_trace_id` and `ctx_span_id` have been set to integers since the beginning (see lines 196 and 197).
If, as you mentioned, the values coming from the `get_instana_ancestor()` are string, then you have to change them by evaluating their type where `ctx_trace_id` and `ctx_span_id`  are overwritten on lines 227 and 228: 
```
if instana_ancestor:
    ctx_trace_id = int(instana_ancestor.t)
    ctx_span_id = int(instana_ancestor.p)
```

Comment (CagriYonca) [reply src/instana/propagators/base_propagator.py:264]: get_traceparent_fields is also returning string below, we'll need to cast to integer it's values as well, you can see the tp_trace_id and tp_parent_id are being set to ctx_trace_id and ctx_span_id in disable_traceparent block:


 ```
  _, tp_trace_id, tp_parent_id, _ = self._tp.get_traceparent_fields(
        traceparent
    )

    if tracestate and "in=" in tracestate:
        instana_ancestor = self._ts.get_instana_ancestor(tracestate)

    if disable_traceparent == "":
        ctx_trace_id = tp_trace_id[-16:]
        ctx_span_id = tp_parent_id
        ctx_synthetic = synthetic
        ctx_trace_parent = True
        ctx_instana_ancestor = instana_ancestor
        ctx_long_trace_id = tp_trace_id
    else:
        if instana_ancestor:
            ctx_trace_id = int(instana_ancestor.t)
            ctx_span_id = int(instana_ancestor.p)
            ctx_synthetic = synthetic

```

Comment (pvital) [reply src/instana/propagators/base_propagator.py:264]: > get_traceparent_fields is also returning string below, we'll need to cast to integer it's values as well, you can see the tp_trace_id and tp_parent_id are being set to ctx_trace_id and ctx_span_id in disable_traceparent block:

So, go ahead and do it there as well! 😉

Comment (CagriYonca) [reply src/instana/propagators/base_propagator.py]: Done 👍

File: src/instana/collector/base.py [hunk 2]
```diff
@@ -150,12 +150,6 @@ def background_report(self):
 
         self.prepare_and_report_data()
 
-        if self.thread_shutdown.is_set():
-            logger.debug(
-                "Thread shutdown signal is active: Shutting down reporting thread"
-            )
-            return False
-
```
Comment (GSVarsha) [inline src/instana/collector/base.py:158]: Any specific reason to remove this?

File: src/instana/span_context.py [hunk 3]
```diff
@@ -1,6 +1,7 @@
 # (c) Copyright IBM Corp. 2021
 # (c) Copyright Instana Inc. 2019
 
+from opentelemetry.trace import SpanContext as OtelSpanContext
```
Comment (GSVarsha) [inline src/instana/span_context.py:4]: Import already available [here](https://github.com/instana/python-sensor/pull/603/files#diff-56b8f1677bf62524a185ea8a6e852056cfadb96a983b0270772abb92cb39c5cfR8).

File: src/instana/span.py
Comment (GSVarsha) [inline src/instana/span.py:1]: This file is outdated. Please take a look at `src/instana/span` folder if you want to make some changes.

**Review by GSVarsha — CHANGES_REQUESTED** [review]
Few requests ⬆️

File: src/instana/span.py
Comment (pvital) [reply src/instana/span.py:1]: 👆🏻 this!

File: tests/collector/__init__.py
Comment (pvital) [inline tests/collector/__init__.py:1]: You are moving/renaming the `__init__.py` file from the old `platforms` directory to the `collector`, and letting the new `agents` directory without an init file.

I suggest you to:
- First, rename the new `agents` dir name to `agent` - coherent with the same name in `src/instana/` 
- Second, create a new `__init__.py` file in the `tests/collector` folder and move this one to the new `agent` dir.

File: tests/util/test_traceutils.py [hunk 4]
```diff
@@ -0,0 +1,65 @@
...
+    tracing_is_off,
+)
+from instana.singletons import agent, tracer
+
+
+class TestTraceUtils:
+    @pytest.fixture(autouse=True)
+    def _resource(self):
+        pass
+
```
Comment (pvital) [inline tests/util/test_traceutils.py:18]: No need to have this if you are not setting up and tearing down the test execution.

File: tests/util/test_traceutils.py [hunk 5]
```diff
@@ -0,0 +1,65 @@
...
+    def test_extract_custom_headers(self, span) -> None:
```
Comment (pvital) [inline tests/util/test_traceutils.py:19]: Missing type annotation.

**Review by pvital — CHANGES_REQUESTED** [review]
Few requests

File: src/instana/collector/base.py (hunk 2 above)
Comment (CagriYonca) [reply src/instana/collector/base.py:145]: It was removed by a mistake, re-added it
